From a066ce376c783327f3842598cfd624f6ce6a0b80 Mon Sep 17 00:00:00 2001
From: "yi.zhang1" <yi.zhang1@amlogic.com>
Date: Wed, 10 May 2023 17:41:12 +0800
Subject: [PATCH] amlbian: Optimize gstreamer compilation  [1/1]

PD#SWPL-114580

Problem:
amlbian gstreamer build,apply patch fail.

Solution:
Optimize the compilation process.

Verify:
local

Signed-off-by: yi.zhang1 <yi.zhang1@amlogic.com>
Change-Id: Ic11e7769776988108683ed16547faaf005b9ab4c
---
 config/functions/build-gstreamer | 94 +++++++++++++-------------------
 1 file changed, 37 insertions(+), 57 deletions(-)

diff --git a/config/functions/build-gstreamer b/config/functions/build-gstreamer
index 45b3c0d..d2cd470 100755
--- a/config/functions/build-gstreamer
+++ b/config/functions/build-gstreamer
@@ -1,75 +1,59 @@
 source config/functions/common-functions
+
 GSTPKGNAME="gstreamer-1.20.3"
-export GSTREAMER_COMPILE_DIR=$GSTREAMER_VENDOR_DIR/$GSTPKGNAME
-export GST_BUILD_DIR=$GSTREAMER_COMPILE_DIR/arm64-build
-export GST_GOOD_DEB="${GSTPKGNAME}-plugin-v4l2src"
+GSTREAMER_COMPILE_DIR=$GSTREAMER_VENDOR_DIR/$GSTPKGNAME
+GST_BUILD_DIR=$GSTREAMER_COMPILE_DIR/arm64-build
+GST_GOOD_DEB="${GSTPKGNAME}-plugin-v4l2src"
 
 build_gstreamer() {
+
+
+
+    # already build pass, needn't build again
+    if [ -f "$GSTREAMER_COMPILE_DIR"/.buildpass ]; then
+	echo "gstreamer already build pass"
+	return
+    else
+        rm -rf $GSTREAMER_COMPILE_DIR
+    fi
+
     # copy gstreamersrc to build folder
-	if [ -f "$GSTREAMER_COMPILE_DIR"/.scmversion ]; then
-		echo "gstreamer already exist, needn't copy"
-	else
-		echo "cp -rf "$PKG_BUILD_CACHE"/$GSTPKGNAME/ $GSTREAMER_VENDOR_DIR"
-		eval "cp -rf "$PKG_BUILD_CACHE"/$GSTPKGNAME/ $GSTREAMER_VENDOR_DIR"
-	fi
-
-    if [ -f "$GSTREAMER_COMPILE_DIR"/.cachefile ]
+    echo "cp -rf "$PKG_BUILD_CACHE"/$GSTPKGNAME/ $GSTREAMER_VENDOR_DIR"
+    eval "cp -rf "$PKG_BUILD_CACHE"/$GSTPKGNAME/ $GSTREAMER_VENDOR_DIR"
+
+
+    if [[ -d "$PKG_BUILD_CACHE"/gstreamer_cache && -f "$PKG_BUILD_CACHE"/gstreamer_cache/.cachesuccess ]]
     then
-        echo "gstreamer cache file already exist,needn't copy"
+        echo "cp -rf "$PKG_BUILD_CACHE"/gstreamer_cache/ $GSTREAMER_COMPILE_DIR/subprojects"
+        eval "cp -rf "$PKG_BUILD_CACHE"/gstreamer_cache/* $GSTREAMER_COMPILE_DIR/subprojects"
     else
-        if [[ -d "$PKG_BUILD_CACHE"/gstreamer_cache && -f "$PKG_BUILD_CACHE"/gstreamer_cache/.cachesuccess ]]
-        then
-            echo "cp -rf "$PKG_BUILD_CACHE"/gstreamer_cache/ $GSTREAMER_COMPILE_DIR/subprojects"
-            eval "cp -rf "$PKG_BUILD_CACHE"/gstreamer_cache/* $GSTREAMER_COMPILE_DIR/subprojects"
-            gst_cache_list_before_install=`ls $GSTREAMER_COMPILE_DIR/subprojects`
-            touch $GSTREAMER_COMPILE_DIR/.cachefile
-        else
-            info_msg "build_gst_cache_prepare start"
-            mkdir -p $PKG_BUILD_CACHE/gstreamer_cache
-            gst_cache_list_before_install=`ls $GSTREAMER_COMPILE_DIR/subprojects`
-
-            echo "gst_cache_list_before_install:" $gst_cache_list_before_install
-        fi
+        info_msg "build_gst_cache_prepare start"
+        mkdir -p $PKG_BUILD_CACHE/gstreamer_cache
     fi
 
+    gst_cache_list_before_install=`ls $GSTREAMER_COMPILE_DIR/subprojects`
+    echo "gst_cache_list_before_install:" $gst_cache_list_before_install
 
 	cd "$GSTREAMER_COMPILE_DIR"
-	touch .scmversion
 
-	# already build pass, needn't build again
-	if [ -f "$GSTREAMER_COMPILE_DIR"/.buildpass ]; then
-		echo "gstreamer already build pass"
-		return
-	fi
 
     #for aml patch to use
     info_msg "[Gstreamer]apply patch ..."
     GST_GOOD_PATCH_LIST=`ls ${PATCH_DIR}/gst-plugins-good-patchs`
 
-    if [ -f "$GSTREAMER_COMPILE_DIR"/.patchsuccess ]; then
-         echo "already patched"
-    else
-        for GST_GOOD_PATCH in $GST_GOOD_PATCH_LIST
-        do
-            echo "need patch"
-            echo "$GST_GOOD_PATCH"
-            GST_GOOD_PATCH_DIR=${PATCH_DIR}/"gst-plugins-good-patchs/$GST_GOOD_PATCH"
-            do_patch "$GST_GOOD_PATCH_DIR"
-            touch .patchsuccess
-        done
-    fi
+    for GST_GOOD_PATCH in $GST_GOOD_PATCH_LIST
+    do
+        echo "need patch"
+        echo "$GST_GOOD_PATCH"
+        GST_GOOD_PATCH_DIR=${PATCH_DIR}/"gst-plugins-good-patchs/$GST_GOOD_PATCH"
+        do_patch "$GST_GOOD_PATCH_DIR"
+    done
+
 
     info_msg "[Gstreamer]meson build ninja..."
 
     cd "$GSTREAMER_COMPILE_DIR"
 
-    if [ ! -f "$GSTREAMER_COMPILE_DIR /meson.build" ]; then
-        echo "need meson.build file"
-     #   meson build
-    else
-        echo "meson.build already exit"
-    fi
-
     #amlbian gstreamer build need yocto .so file
     info_msg "config the env first"
     if [ ! -f "/opt/gst-libs" ]; then
@@ -86,12 +70,8 @@ build_gstreamer() {
     then
         echo "cp $ROOT/arm64.txt $GSTREAMER_COMPILE_DIR"
         cp $ROOT/arm64.txt $GSTREAMER_COMPILE_DIR
-        if [ ! -f ".configsuccess" ]
-        then
-            echo "config the arm64 toolchain and env"
-            meson --cross-file arm64.txt arm64-build --prefix=/usr/lib --libdir=/usr/lib/aarch64-linux-gnu --libexecdir=gstreamer1.0
-            touch .configsuccess
-        fi
+        echo "config the arm64 toolchain and env"
+        meson --cross-file arm64.txt arm64-build --prefix=/usr/lib --libdir=/usr/lib/aarch64-linux-gnu --libexecdir=gstreamer1.0
     fi
 
     #cp the download file to the cache
@@ -142,7 +122,7 @@ clean_gstreamer() {
         echo " do ninja clean "
         cd $GST_BUILD_DIR
         ninja clean
-        rm $GSTREAMER_COMPILE_DIR/.buildpass
+        rm -rf $GSTREAMER_COMPILE_DIR/.buildpass
 }
 
 if [ "$DISTRIB_ARCH" == arm64 ]; then
-- 
2.25.1

